src/language-js/print/type-annotation.js

```diff
@@ -321,6 +296,84 @@ function printJSDocType(path, print, token) {
   ];
 }
 
+function printTypeAnnotationProperty(path, options, print) {
+  const {
+    node: { typeAnnotation },
+  } = path;
+  if (!typeAnnotation) {
+    return "";
+  }
+
+  let shouldPrintLeadingSpace = false;
+
+  if (
+    (typeAnnotation.type === "TSTypeAnnotation" ||
+      typeAnnotation.type === "TypeAnnotation") &&
+    hasComment(typeAnnotation, CommentCheckFlags.leading) &&
```

Comment:
```suggestion
    hasComment(typeAnnotation, CommentCheckFlags.Leading) &&
```

Reply:
Add `/** @enum {number} */` in front of `const CommentCheckFlags = {` for TS to catch this.

Reply:
Thanks, but I think I need find a better way to do this, it can exist in other properties, had to cover all cases, maybe use `willPrintOwnComment`.